server/middleware/users.ts

```diff
@@ -0,0 +1,21 @@
+import { ForbiddenError } from 'express-response-errors';
+import DOMpurify from 'isomorphic-dompurify';
+
+import { AnyRequestHandler } from 'server/types/express';
+
+export const requireSameId: AnyRequestHandler = async (req, res, next) => {
+  if (req.user.dataValues.id.toString() !== req.params.id) {
+    throw new ForbiddenError('User can only make changes to their own profile');
+  }
+
+  next();
+};
+
+export const removeMarkup: AnyRequestHandler = async (req, res, next) => {
+  const fields = Object.keys(req.body);
+  fields.forEach((field) => {
+    req.body[field] = DOMpurify.sanitize(req.body[field]);
+  });
+
+  next();
+};
```

Comment:
we can test middleware too! no server/`TestServer` instance needed, you can just test it as if it were any old function - just pass a (req with a body, empty object res, and an unimplemented [jest mock function](https://jestjs.io/docs/jest-object#jestfnimplementation) for next), call the middleware, and you can verify that `req.body` should've been mutated to account for it + the next function was called